app/src/main/java/com/infomaniak/swisstransfer/ui/components/NewTransferFab.kt

```diff
@@ -20,24 +20,31 @@ package com.infomaniak.swisstransfer.ui.components
 import androidx.compose.material3.FloatingActionButtonDefaults
 import androidx.compose.material3.FloatingActionButtonElevation
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
-import com.infomaniak.swisstransfer.ui.NewTransferActivity
-import com.infomaniak.swisstransfer.ui.utils.launchActivity
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.UploadSourceChoiceBottomSheet
 
 @Composable
 fun NewTransferFab(
     modifier: Modifier = Modifier,
     newTransferFabType: NewTransferFabType
 ) {
-    val context = LocalContext.current
+    var showUploadSourceChoiceBottomSheet by rememberSaveable { mutableStateOf(false) }
+
     SwissTransferFab(
         modifier = modifier,
         fabType = newTransferFabType.fabType,
         elevation = newTransferFabType.elevation(),
-        onClick = { context.launchActivity(NewTransferActivity::class) },
+        onClick = { showUploadSourceChoiceBottomSheet = true },
+    )
+    UploadSourceChoiceBottomSheet(
```

Comment:
Maybe we can move the `if` outside `UploadSourceChoiceBottomSheet` ? Because it's weird that `UploadSourceChoiceBottomSheet` is handling it's own visibility.